core/authentication_api.php

```diff
@@ -764,29 +783,29 @@ function auth_does_password_match( $p_user_id, $p_test_password ) {
  * @access public
  */
 function auth_process_plain_password( $p_password, $p_salt = null, $p_method = null ) {
-	$t_login_method = config_get( 'login_method' );
-	if( $p_method !== null ) {
-		$t_login_method = $p_method;
-	}
+	$t_login_method = $p_method !== null ? $p_method : config_get( 'login_method' );
 
 	switch( $t_login_method ) {
 		case CRYPT:
-
 			# a null salt is the same as no salt, which causes a salt to be generated
 			# otherwise, use the salt given
 			$t_processed_password = crypt( $p_password, $p_salt );
 			break;
 		case MD5:
 			$t_processed_password = md5( $p_password );
 			break;
+		case SAFE_HASH:
+			$t_processed_password = password_hash( $p_password, PASSWORD_DEFAULT );
```

Comment:
Using the default allows the same code base to automatically use the most advanced available algorithm, which I think is nice. 

I thought about the database field size given the recommendation in the PHP manual, but given how much advance notice we'd get before a change of default algo would be effective, I think it would be a better approach to make a schema change (can be done now or later) to increase the password field's size.

>  The documentation recommends 255 and we are currently 64. Not sure if we can move to 191 vs. 255. The bcrypt requires 60 characters, so we are covered there.

I know, I had the BLOWFISH hash length in mind when I increased the field size back in 2015.

The 191 chars limit only applies to MySQL when the column is indexed, which is not the case of the _password_ field. So there should be no issue with using 255 here.